lib/dispatch.c

```diff
@@ -73,7 +73,7 @@ DispatchStatus dispatch_run(DispatchPtr table, int timeout) {
 			if (errno != EINTR) {
 				log_error("Error waiting for I/O events");
 			}
-			return 0;
+			return 1;
```

Comment:
Is this a fix for a bug (i.e. an incorrect return value), or what's the purpose of this change? :thinking:

Reply:
`poll` will be interrupted by the SIGHUP as well, and returning `0` here causes the main program to break out of the loop at and terminate.

https://github.com/onitake/daliserver/blob/4cfdb984e9e054d9e9008815d4253c9ba551fef3/src/daliserver.c#L180

If instead, `1` is returned here, the same loop in the main program will consider the value of `running` as well, which will still be `1` after a SIGHUP, but `0` after a SIGINT/SIGTERM.